docs/UserGuide.adoc

```diff
@@ -377,57 +412,83 @@ TODO: ADD DETAILS ONCE IMPLEMENTED
 [[FilterStudent]]
 ==== Filter `filter`
 
-TODO: ADD DETAILS ONCE IMPLEMENTED
+Filter Student returns you all the students according to the given keywords below.
+
+You can filter students in a class using these two ways:
+
+****
+1. module code and group code `m/` `g/`
+2. module code `m/`
+****
+
+1. Using both module code and group code returns you all the students specified by the group code and module code.
+
+Format: `student filter [m/MOD_CODE] g/GROUP_CODE`
+
+Examples:
+
+* ` student filter m/cs2103t g/g06`
+Filters all students in module `CS2103T`, under group `G06`.
+
+2. Using just the module code will return you all the students from the first group of module code shown in UI.
```

Comment:
Using just the module code will show you the students inside the first group of the module specified by the given module code.